Content.Server/Delivery/DeliverySystem.cs

```diff
@@ -0,0 +1,90 @@
+using Content.Server.Cargo.Components;
+using Content.Server.Cargo.Systems;
+using Content.Server.Station.Systems;
+using Content.Server.StationRecords.Systems;
+using Content.Shared.Delivery;
+using Content.Shared.FingerprintReader;
+using Content.Shared.Hands.EntitySystems;
+using Content.Shared.Labels.EntitySystems;
+using Content.Shared.NameModifier.EntitySystems;
+using Content.Shared.Popups;
+using Content.Shared.StationRecords;
+using Content.Shared.Tag;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Containers;
+
+namespace Content.Server.Delivery;
+
+/// <summary>
+/// System for managing deliveries spawned by the mail teleporter.
+/// This covers for mail spawning, as well as granting cargo money.
+/// </summary>
+public sealed partial class DeliverySystem : SharedDeliverySystem
+{
+    [Dependency] private readonly CargoSystem _cargo = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly StationRecordsSystem _records = default!;
+    [Dependency] private readonly StationSystem _station = default!;
+    [Dependency] private readonly FingerprintReaderSystem _fingerprintReader = default!;
+    [Dependency] private readonly SharedLabelSystem _label = default!;
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<DeliveryComponent, MapInitEvent>(OnMapInit);
+
+        InitializeSpawning();
+    }
+
+    private void OnMapInit(Entity<DeliveryComponent> ent, ref MapInitEvent args)
+    {
+        _container.EnsureContainer<Container>(ent, ent.Comp.Container);
+
+        var stationId = _station.GetStationInMap(Transform(ent).MapID);
+
+        if (stationId == null)
+            return;
+
+        _records.TryGetRandomRecord<GeneralStationRecord>(stationId.Value, out var entry);
+
+        if (entry == null)
+            return;
+
+        ent.Comp.RecipientName = entry.Name;
+        ent.Comp.RecipientJobTitle = entry.JobTitle;
+        ent.Comp.RecipientStation = stationId.Value;
+
+        _appearance.SetData(ent, DeliveryVisuals.JobIcon, entry.JobIcon);
+
+        _label.Label(ent, ent.Comp.RecipientName);
+
+        if (TryComp<FingerprintReaderComponent>(ent, out var reader) && entry.Fingerprint != null)
+        {
+            _fingerprintReader.AddAllowedFingerprint((ent.Owner, reader), entry.Fingerprint);
+            Dirty(ent, reader);
```

Comment:
```suggestion
            _fingerprintReader.AddAllowedFingerprint((ent.Owner, reader), entry.Fingerprint);
```
`AddAllowedFingerprint` calls dirty itself.